### PR TITLE
added fetch by id for course

### PR DESCRIPTION
**Pull request by SanaBob**

### CODE REVIEW

File: app/views/course.py [hunk 1]
```diff
@@ -1,9 +1,25 @@
...
+
+    def get_queryset(self):
+        course_id = self.request.query_params.get("id", None)
+        if course_id:
+            get_object_or_404(Course, pk=course_id)
+            return self.queryset
+        return self.queryset
+
+    def get_serializer_class(self):
+        course_id = self.request.query_params.get("id", None)
```
Comment (Opeyem1a) [inline app/views/course.py:22]: idk how much I like this, it basically means if we ever filter by course is will use the limited version of the serializer.

Why did we need the 2nd serializer in the first place btw? If the extra data returned in a course is not significant, we could have just used the existing one

File: app/views/course.py [hunk 2]
```diff
@@ -1,9 +1,25 @@
...
+from app.serializers.course import CourseSerializer, CustomCourseSerializer
 
 
 class CourseViewSet(viewsets.ModelViewSet):
     queryset = Course.objects.all()
-    serializer_class = CourseSerializer
+    filter_backends = [filters.DjangoFilterBackend]
+    filterset_fields = ["id"]
+
+    def get_queryset(self):
```
Comment (Opeyem1a) [inline app/views/course.py:14]: why did we need to define this function? Would it have already 404'd without it?

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

File: app/views/course.py (hunk 2 above)
Comment (SanaBob) [reply app/views/course.py:14]: It returns an empty list when there is no match

File: app/views/course.py (hunk 1 above)
Comment (SanaBob) [reply app/views/course.py:22]: We can just use the 1st serializer, but I thought the spec only wanted limited data on specific search param courseId only, and this happen to be the only way I found possible

**Comment by SanaBob** [comment]
I didn't know that search by parameter was in default view, but I have verified that it works

File: app/serializers/course.py [hunk 3]
```diff
@@ -1,12 +1,9 @@
 from rest_framework import serializers
 
 from app.models.course import Course
-from app.serializers.attribute import AttributeSerializer
 
 
 class CourseSerializer(serializers.ModelSerializer):
-    attributes = AttributeSerializer(many=True, read_only=True)
```
Comment (Opeyem1a) [inline app/serializers/course.py:8]: I think you misunderstood my comment. I was just challenging that if the difference between your new serializer and the old one is trivial, don't make a new one. In this case, this is not really trivial to just delete this part from the serializer

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

**Pull request closed without merging**